### PR TITLE
(#170766098) Feature to create, update, retrieve and delete a comment on travel request

### PR DESCRIPTION
**Pull request by jajajaden01**
### What does this PR do?
- To create a trip comment, and able to delete that comment

### Description of the task to be completed
- create add and delete comment functionalities.
- testing both endpoints

### How should this be manually tested?
 - clone the repository, run the server locally then follow the instructions from the swagger documentation.

### What are the relevant pivotal tracker stories?
[#170766098](https://www.pivotaltracker.com/story/show/170766098)

### CODE REVIEW

File: server/controllers/TripCommentController.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import tripHelpers from '../helpers/tripHelpers';
+import commentHelper from '../helpers/TripCommentHelper';
+import checkIdParams from '../helpers/CheckParams';
+
+/**
+ * This class contains all methods
+ * required to handle
+ * comment endpoints' request.
+ */
+class TripController {
```
Comment (istafrica-dennw) [inline server/controllers/TripCommentController.js]: @jajajaden01 is this TripController? or commentController?

File: server/controllers/TripCommentController.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    const tripExists = await tripHelpers.userTripExists(req.params.tripId, req.user.id);
+    if (!tripExists) {
+      return res.status(404).json({
+        status: res.statusCode,
+        error: 'Sorry, This trip does not exist.'
+      });
+    }
+
+    const savedComment = await commentHelper.saveComment(req.body);
```
Comment (istafrica-dennw) [inline server/controllers/TripCommentController.js:36]: can this line 36 fail? what happens when it does? to the code block 38 - 42 ?

File: server/middlewares/TripCommentValidator.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ */
+class CommentValidator {
+  /**
+   * This method handle the signup request.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @param {object} next The response.
+   * @returns {object} The status and some data of the user.
+   */
+  static NewComment(req, res, next) {
```
Comment (istafrica-dennw) [inline server/middlewares/TripCommentValidator.js]: is this a function, a constant, a class ?. Make sure you are following best practices for function naming. Thank you

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

File: server/controllers/TripCommentController.js (hunk 1 above)
Comment (jajajaden01) [reply server/controllers/TripCommentController.js]: i'm going to correct that

File: server/controllers/TripCommentController.js (hunk 2 above)
Comment (jajajaden01) [reply server/controllers/TripCommentController.js:36]: If it fails, we get the server response (500) in a catch(){...}
in the route, there is a common function we use for try and catch

File: server/middlewares/TripCommentValidator.js (hunk 3 above)
Comment (jajajaden01) [reply server/middlewares/TripCommentValidator.js]: this was a mistake, I'm going to change it.

**Pull request closed without merging**